sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -2649,3 +2652,155 @@ case class Sentences(
     copy(str = newFirst, language = newSecond, country = newThird)
 
 }
+
+case class ToPrettyString(child: Expression, timeZoneId: Option[String] = None)
+  extends CastBase {
+  import ToPrettyString._
+
+  override def withTimeZone(timeZoneId: String): TimeZoneAwareExpression =
+    copy(timeZoneId = Option(timeZoneId))
+
+  override def dataType: DataType = StringType
+
+  private lazy val timeFormatters: TimeFormatters =
+    TimeFormatters(DateFormatter(zoneId), TimestampFormatter.getFractionFormatter(zoneId))
+
+  override def leftBracket: String = "{"
+  override def rightBracket: String = "}"
+  override def elementSpace: String = ""
+  override def keyValueSeparator: String = ":"
+  override def structTypeWithSchema: Boolean = true
+
+  override def nullSafeEval(input: Any): Any = {
+    UTF8String.fromString(toHiveString((input, child.dataType), false, timeFormatters))
```

Comment:
Since we can reuse the CAST code, shouldn't we just do
```
private lazy val toStringFunc = cast(child.dataType, StringType)

override def nullSafeEval(input: Any): Any = toStringFunc(input)
```

Reply:
Actually we don't need to override `nullSafeEval`. `CastBase.nullSafeEval` already works here.

Reply:
Year,  just override `castToString` and `castToStringCode`